### ISSUE TITLE
How to set up GitHub OAuth for both local development and production? 

### ISSUE THREAD

**Issue by jcohenho** [nextauthjs/next-auth#7581]
### Question 💬

Github OAuth apps only allow for a [single callback URL](https://next-auth.js.org/providers/github). How can we configure our Next.js apps to allow for a single GitHub app to be configured for both development and production environments? Is the best solution to create two different GitHub apps for each environment? 

### How to reproduce ☕️

See question above, and relevant documentation [here](https://next-auth.js.org/providers/github). 

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by balazsorban44** [comment]
The best solution is indeed to create two separate clients. It is good practice to separate testing/production environments anyway.

**Comment by bcjordan** [comment]
Is there any way to incorporate a GitHub App instead of Oauth into NextAuth, and later request more fine-grained tokens for accessing repositories? The Oauth-only flow seems to only allow for global access to all repositories (which users often do not like when first trying out an app)

**Comment by goerlitz** [comment]
I ran into the same issue that the Oauth app allows only 1 callback URL.
After some testing, I figured that both Oauth app and Github App work, but
1. For Oauth app I basically need to create separate app with different callback URLs for my `local`, `dev` and `production` environment. Actually, I don't see any real benefits in managing separate github Oauth app configs for testing+prod. Yes, the environments are separated, but they could share the same github client Id and secret.
2. For Github app I can enter multiple callback URL covering all environments.
